cmake/onnxruntime_providers.cmake

```diff
@@ -369,13 +391,15 @@ if (onnxruntime_USE_TENSORRT)
   file(GLOB_RECURSE onnxruntime_providers_tensorrt_cc_srcs CONFIGURE_DEPENDS
     "${ONNXRUNTIME_ROOT}/core/providers/tensorrt/*.h"
     "${ONNXRUNTIME_ROOT}/core/providers/tensorrt/*.cc"
+    "${ONNXRUNTIME_ROOT}/core/providers/shared_library/*.h"
+    "${ONNXRUNTIME_ROOT}/core/providers/shared_library/*.cc"
```

Comment:
I don't think this is right.

Reply:
What part of it is wrong? The providers need to include the provider side of the shared library interface and that's what this is doing (dnnl did this the same way)